public/app/features/logs/components/LogDetailsRow.tsx

```diff
@@ -160,14 +179,32 @@ class UnThemedLogDetailsRow extends PureComponent<Props, State> {
     showFieldsStats: false,
     fieldCount: 0,
     fieldStats: null,
+    showLineScroll: false,
   };
 
+  logRowRef: React.RefObject<HTMLTableCellElement> = React.createRef();
+
   componentDidUpdate() {
     if (this.state.showFieldsStats) {
       this.updateStats();
     }
   }
 
+  componentDidMount(): void {
+    // convert rem to px
+    const fontSizePx =
+      parseFloat(this.props.theme.typography.bodySmall.fontSize) *
+      parseFloat(getComputedStyle(document.documentElement).fontSize);
+    const parsedValWidth = measureText(this.props.parsedValue, fontSizePx).width;
+    const refWidth = this.logRowRef.current?.clientWidth;
+    const linesBeforeScroll = 3;
```

Comment:
This is loosely tied to the magic number in the css for how many lines to display in the scrollable, I can make this a class variable instead, let me know :)